Documentation/driver-api/virtual_bus.rst

```diff
@@ -0,0 +1,164 @@
+===========
+Virtual Bus
+===========
+
+See <linux/virtual_bus.h> for the models for virtbus_device and virtbus_driver.
+
+In some subsystems, the functionality of the core device (PCI/ACPI/other) may
+be too complex for a single device to be managed as a monolithic block or
+a part of the functionality might need to be exposed to a different subsystem.
+Splitting the functionality into smaller orthogonal devices would make it
+easier to manage data, power management and domain-specific communication with
+the hardware. A key requirement for such a split is that there is no dependency
+on a physical bus, device, register accesses or regmap support. These
+individual devices split from the core cannot live on the platform bus as they
+are not physical devices that are controlled by DT/ACPI. The same argument
+applies for not using MFD in this scenario as it relies on individual function
+devices being physical devices that are DT enumerated.
+
+An example for this kind of requirement is the audio subsystem where a single
+IP handles multiple entities such as HDMI, Soundwire, local devices such as
+mics/speakers etc. The split for the core's functionality can be arbitrary or
+be defined by the DSP firmware topology and include hooks for test/debug. This
+allows for the audio core device to be minimal and tightly coupled with
+handling the hardware-specific logic and communication.
+
+The virtual bus is aimed at being a minimal software bus where virtbus
+devices carved out to represent a part of their core device's functionality can
+live on. This bus is also meant to be generic and be available to be used by
+any type of device (PCI/ACPI/DT). This means that virtbus devices created by
+different subsystems may co-exist on the same bus without sharing much in
+common with each other. Only the devices created by the same device share a
+common parent and establish a shared communication. The minimalist
+implementation for virtual bus also allows for extensibility by making
+specialized use of the virtbus_device by encapsulating it within other
+domain-specific structures.
```

Comment:
Maybe reverse the flow:

The virtual bus is intended to be minimal, generic and avoid domain-specific assumptions. Each virtual bus device represents a part of its parent functionality. The generic behavior can be extended and specialized as needed by encapsulating a virtual bus device within other domain-specific structures and the use of .ops callbacks. Devices on the same virtual bus do not share any structures and the use of a communication channel with the parent is domain-specific.